crates/bevy_remote/src/builtin_methods.rs

```
@@ -786,30 +791,90 @@ pub fn process_remote_query_request(In(params): In<Option<Value>>, world: &mut W
     let mut response = BrpQueryResponse::default();
     let mut query = query.build();
     for row in query.iter(world) {
-        // The map of component values:
-        let components_map = build_components_map(
-            row.clone(),
-            paths_and_reflect_components.iter().copied(),
-            &type_registry,
-        )
-        .map_err(BrpError::component_error)?;
+        let entity_id = row.id();
+        let entity_ref = world.get_entity(entity_id).expect("Entity should exist");
+
+        let components_map = if components.is_empty() {
```

Should it really include all components when `components: []`?

Receiving all component data seems like something that is only useful in certain edge cases like taking snapshots/dumps or replicating world state between apps.

Is it possible to make query that won't include _any_ component data with the new implementation? E.g. something like: `Query<Entity, With<MyComponent>>`

Maybe something like `components: 'all'` to explicitly opt-in?

---

As discussed on discord, there are downsides to doing this:

1. This requires a modification to the existing API
2. There was contention previously with creating a separate API to query all components
3. You lose the implicit benefit of being able to get back custom reflected `Component`s and their values by default without having to make another REST call to retrieve them.

You also had mentioned that you had an example query like below:
```
data: {
  option: [
    bevyTypes.ChildOf,
    bevyTypes.Name,
  ],
  has: [
    bevyTypes.Camera2d,
    bevyTypes.Camera3d,
    bevyTypes.DirectionalLight,
    bevyTypes.Mesh2d,
    bevyTypes.Mesh3d,
    bevyTypes.Node,
    bevyTypes.PointLight,
    bevyTypes.SpotLight,
    bevyTypes.Sprite,
    bevyTypes.Button,
    bevyTypes.Label,
    bevyTypes.UiImage,
    bevyTypes.Text,
    bevyTypes.TextSpan,
    bevyTypes.Transform,
    bevyTypes.PrimaryWindow,
    bevyTypes.Window,
    bevyTypes.PointerId,
  ],
},
```

I'm trying to explicitly avoid having to know all the possible types in the inspector, especially when trying to build the tree for the first time. It is possible to get this data, but again you need multiple rest calls.

---

I think there might be slight issues with conflicting queries that both specify to get all components (`components: []` as in the current implementation) at the same time as requesting `option` and `has` components: what's the point? In my opinion a good implementation would make conflicting queries impossible to formulate.

To address this I would consider another option: a new boolean field.

You could either query for all components:

```json
{
  "jsonrpc": "2.0", "id": 0, "method": "bevy/query", "params": {
    "data": {
      "allComponents": true
    }
  }
}
```

Or a subset:

```json
{
  "jsonrpc": "2.0", "id": 0, "method": "bevy/query", "params": {
    "data": {
      "allComponents": false,
      "components": ["..."],
      "option": ["..."],
      "has": ["..."]
    }
  }
}
```

With `"allComponents": false` being optional since it would be implied/default.

Then maybe throw an error if `allComponents: true` and other fields are not null?

---

@splo that wouldn't be possible. The type path at the very least includes the crate name, so it would be `foo::all`.